cpp/src/slice2cpp/Gen.cpp

```diff
@@ -72,20 +72,21 @@ namespace
         }
     }
 
-    string getDeprecateSymbol(const ContainedPtr& p1, const ContainedPtr& p2)
+    string getDeprecateSymbol(const ContainedPtr& p1)
     {
-        string deprecateMetadata, deprecateSymbol;
-        if (p1->findMetaData("deprecate", deprecateMetadata) ||
-            (p2 != 0 && p2->findMetaData("deprecate", deprecateMetadata)))
+        string deprecatedSymbol;
+        if (p1->isDeprecated(true))
         {
-            string msg = "is deprecated";
-            if (deprecateMetadata.find("deprecate:") == 0 && deprecateMetadata.size() > 10)
+            if (auto reason = p1->getDeprecationReason(false))
```

Comment:
Should this be `true` as well? Since we're checking the parent above?

Reply:
Yes, you're right. Fixed.